### PR TITLE
[Teacher][MBL-13157] Add refresh to submissions list when post/hide grades

### PR DESCRIPTION
**Pull request by CalvinKern**
To test:

Post or Hide grades on the new post policy screen in Teacher.
Wait for a few seconds while the delayed job on Canvas finishes.
Verify that the screen will go back and refresh the submission list automagically.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1005484&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/teacher/src/main/java/com/instructure/teacher/features/postpolicies/PostGradeEffectHandler.kt [hunk 1]
```diff
@@ -45,20 +49,51 @@ class PostGradeEffectHandler : EffectHandler<PostGradeView, PostGradeEvent, Post
...
+
+    private suspend fun watchProgress(progressId: String?) {
+        if (progressId == null) { // If we didn't get a progress ID back, then it failed to post
+            consumer.accept(PostGradeEvent.PostFailed)
+            return
+        }
+
+        lateinit var progress: Progress
+        do {
+            progress = ProgressManager.getProgressAsync(progressId).await().dataOrNull ?: run {
```
Comment (JordanMarshall) [inline apps/teacher/src/main/java/com/instructure/teacher/features/postpolicies/PostGradeEffectHandler.kt]: Could we throw a delay in here so we're not hitting the API full throttle? While hiding grades for one of my sandbox assignments the network profiler showed 100+ API calls over ~12 seconds, and that's just with 8 students. We could either do a flat delay of something like 1 second, or use exponential backoff starting at say 500ms and maxing out at maybe 5 or 10 seconds.

Comment (CalvinKern) [reply apps/teacher/src/main/java/com/instructure/teacher/features/postpolicies/PostGradeEffectHandler.kt]: Oh, that's a very good point. I'll get in a backoff thingy. Talking with some network teams to figure out a good max.

**Pull request merged**